components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -637,13 +635,10 @@ compute_filename (const IOFileSpecs& file_specs,
 }
 
 void OutputManager::
-set_params (const ekat::ParameterList& params,
-            const std::map<std::string,std::shared_ptr<fm_type>>& field_mgrs)
+set_params (const std::map<std::string,std::shared_ptr<fm_type>>& field_mgrs)
```

Comment:
I think the only reason we can't run this at init time, is b/c we need the field mgrs. But these are only needed for the restart OM, and only to grab the names of the restart fields. Maybe we should:
- gather the RESTART fields from inside the driver, and set the vector<string> of names in the plist there
- call this method at init time.

This would make the method signature clearer (calling it "set_params" when there's not even a param list in the name is odd, especially since the param list was set in the class long time ago).

Reply:
Yeah, I didn't dive too far in to it, but I see what you are saying. I'll rework it.

Reply:
I just updated the comment and name of `set_params()` -> `setup_internals()` which I don't love, so let me know if you have a better suggestion.

I think you are right that we can preprocess the params before setup, but it wasn't straightforward to me how to best do that since we need to initialize OM before processing FieldRequests, but FM only finalizes it's restart fields after FieldRequests are processed. There probably is a simple solution, but I prefer to merge this and then maybe revisit later.